R/prep_diagram.R

```diff
@@ -2,5 +2,67 @@ prep_diagram <- function(audit_data = NULL, emissions_data = NULL) {
   if (is.null(audit_data) | is.null(emissions_data)) {
     data_out <- use_toy_data("diagram")
   }
-  data_out
+
+  audit_data <- audit_data %>%
+    dplyr::mutate(exposure_portfolio = sum(.data$value_usd, na.rm = TRUE)) %>%
+    dplyr::group_by(.data$asset_type) %>%
+    dplyr::mutate(
+      exposure_asset_class = sum(.data$value_usd, na.rm = TRUE),
+      exposure_asset_class_perc = .data$exposure_asset_class / .data$exposure_portfolio
+    ) %>%
+    dplyr::ungroup() %>%
+    dplyr::mutate(
+      covered_pacta = dplyr::if_else(
+        .data$financial_sector %in% c("Automotive", "Coal", "Oil&Gas", "Power", "Aviation", "Cement", "Steel"),
+        TRUE,
+        FALSE
+      )
+    ) %>%
+    dplyr::group_by(
+      .data$asset_type, .data$exposure_portfolio, .data$covered_pacta,
+      .data$exposure_asset_class, .data$exposure_asset_class_perc
+    ) %>%
+    dplyr::summarise(
+      exposure_pacta = sum(.data$value_usd, na.rm = TRUE),
+      .groups = "drop"
+    ) %>%
+    dplyr::mutate(
+      exposure_pacta_perc_asset_class_exposure = .data$exposure_pacta / .data$exposure_asset_class
+    ) %>%
+    dplyr::ungroup() %>%
+    dplyr::filter(.data$covered_pacta) %>%
+    dplyr::select(
+      c(exposure_portfolio, asset_type, exposure_asset_class,
+        exposure_asset_class_perc, exposure_pacta, exposure_pacta_perc_asset_class_exposure)
+    )
+
+  emissions_data <- emissions_data %>%
+    dplyr::filter(.data$entity == "portfolio") %>%
+    dplyr::group_by(.data$asset_type) %>%
+    # TODO: check if this what we need or if it needs to be divivded by the asset value covered
+    dplyr::mutate(emissions_asset = sum(.data$weighted_sector_emissions, na.rm = TRUE)) %>%
+    dplyr::ungroup() %>%
+    dplyr::mutate(
+      covered_pacta = dplyr::if_else(
+        .data$sector %in% c("Automotive", "Coal", "Oil&Gas", "Power", "Aviation", "Cement", "Steel"),
+        TRUE,
+        FALSE
+      )
+    ) %>%
+    dplyr::group_by(.data$asset_type, .data$covered_pacta, .data$emissions_asset) %>%
+    dplyr::summarise(
+      emissions_pacta = sum(.data$weighted_sector_emissions, na.rm = TRUE),
+      .groups = "drop"
+    ) %>%
+    dplyr::ungroup() %>%
+    dplyr::mutate(emissions_pacta_perc = .data$emissions_pacta / .data$emissions_asset) %>%
+    dplyr::filter(.data$covered_pacta) %>%
+    dplyr::select(c(asset_type, emissions_pacta_perc, emissions_pacta))
+
+  data_out <- audit_data %>%
+    dplyr::inner_join(emissions_data, by = "asset_type") %>%
+    dplyr::rename(asset_class = asset_type) %>%
```

Comment:
```suggestion
    dplyr::rename(asset_class = .data$asset_type) %>%
```

Reply:
so my understanding of this https://dplyr.tidyverse.org/articles/programming.html is that rename should use tidyselection, not data masking?

Reply:
you're right, my bad.... but then wrap `asset_type` in quotes so it doesn't generate notes in R CMD